### PR TITLE
Identity: Rename "namespace" to "type" in the requester interface

### PR DESCRIPTION
**Pull request by ryantxu**
In an effort to avoid confusion with kubernetes namespaces, this PR renames the identity "Namespace" and "NamespaceID" to "Type" and "TypedID" 

Although this PR touches lots of files, there should not be any logic changes

### CODE REVIEW

File: pkg/apimachinery/identity/namespace.go
Comment (ryantxu) [inline pkg/apimachinery/identity/namespace.go:1]: I would like to change the name of this file, but git does not recognize it as a rename -- so lets keep it the same for now and consider renaming later

File: pkg/services/authz/server.go [hunk 1]
```diff
@@ -54,7 +55,16 @@ func (s *legacyServer) Read(ctx context.Context, req *authzv1.ReadRequest) (*aut
 	ctxLogger := s.logger.FromContext(ctx)
 	ctxLogger.Debug("Read", "action", action, "subject", subject, "stackID", stackID)
 
-	permissions, err := s.acSvc.SearchUserPermissions(ctx, stackID, accesscontrol.SearchOptions{NamespacedID: subject, Action: action})
+	var err error
+	opts := accesscontrol.SearchOptions{Action: action}
+	if subject != "" {
+		opts.TypedID, err = identity.ParseTypedID(subject)
```
Comment (ryantxu) [inline pkg/services/authz/server.go:61]: this is a change -- but I think just means we fail sooner on an invalid input

**Review by Jguer — APPROVED** [review]

**Review by gamab — APPROVED** [review]
Fully onboard with that change. It gets super confusing with the token namespace.

**Comment by charandas** [comment]
Likewise. A good refactor to clarify the namespace property as the one for k8s and type for identity type.

**Pull request merged**